src/problem_bank_scripts/tui/tui.py

```diff
@@ -0,0 +1,577 @@
+            variables_str = questionary.text('variables (ex. "mu, sigma")').ask()
+            info["variables"] = variables_str
+            if exercise is not None:
+                if "imports" not in exercise:
+                    exercise["imports"] = []
+                exercise["imports"].append("import prairielearn as pl")
+                exercise["imports"].append("from sympy import sp")
+        case _:
+            print("No other asks for", key)
+    part["info"] = info
+
+
+def extract_variables(text: str, variables: dict) -> str:
+    res = ""
+    open_i = None
+    for i in range(len(text)):
+        if text[i] == "{":
+            open_i = i
+        if open_i is None:  # has to be in the middle
+            res += text[i]
+        if text[i] == "}" and open_i is not None:
+            var = text[open_i+1:i]  # fmt: skip
+            name = ""
+            value = None
+            split = var.split(":")
+            name = split[0].strip()
+            if len(split) > 1:
+                value = split[1].strip()
+            if value:
+                variables[name] = value
+            res += f"{{{{ params.{name} }}}}"
+            open_i = None
+    if open_i is not None:
+        msg = "Unmatched {"
+        raise ValueError(msg)
+    return res.strip()
+
+
+def question_type_from_solution(solution: str) -> str | None:
+    if solution.lower() in ["true", "false"]:
+        return "true-false"
+    if solution.lower() in ["yes", "no"]:
+        return "yes-no"
+    return None
+
+
+def ask_if_not_exists(
+    exercise: dict,
+    key: str,
+    question: str,
+    variables: dict,
+    saved: pathlib.Path,
+    default="",
+    parser=lambda x: x,
+):
+    if key not in exercise:
+        value = parser(questionary.text(question, default=default).ask())
+        if isinstance(value, str):
+            value = extract_variables(value, variables=variables)
+        exercise[key] = value
+        write_json(exercise, saved)
+    return exercise[key]
+
+
+def set_default(exercise: dict, key: str, value: str | list, saved: pathlib.Path):
+    if key not in exercise:
+        exercise[key] = value
+        write_json(exercise, saved)
+    return exercise[key]
+
+
+def run_tui(
+    *,
+    create_pr: bool = False,
+    use_gpt: bool = False,
+    saved: pathlib.Path = pathlib.Path("saved.json"),
+):
+    exercise = {}
+    variables = {}
+    if saved.exists() and questionary.confirm("Would you like to use saved data?").ask():
+        exercise = read_json(saved)
+        if "variables" in exercise:
+            variables = exercise["variables"]
+    try:
+        set_default(exercise, key="assets", value=[], saved=saved)
+
+        textbook = exercise.get("attribution", None)
+        if textbook is None:
+            textbook = questionary.select(
+                message="What textbook is this from? (Select standard if not from any on the list)",
+                choices=KNOWN_ATTRIBUTIONS,
+            ).ask()
```

Comment:
Actually can the default textbook be set with a .env variable?

```suggestion
            textbook = os.environ["TEXTBOOK"] if "TEXTBOOK" in os.environ else questionary.select(
                message="What textbook is this from? (Select standard if not from any on the list)",
                choices=KNOWN_ATTRIBUTIONS,
            ).ask()
```

Reply:
```suggestion
            textbook = os.environ.get("TEXTBOOK") or questionary.select(
                message="What textbook is this from? (Select standard if not from any on the list)",
                choices=KNOWN_ATTRIBUTIONS,
            ).ask()
```